src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -403,10 +402,8 @@ private void HandleSubscription(KafkaConsumerActorMetadata.Internal.ISubscriptio
                         var topicPartitions = assignWithOffset.TopicPartitionOffsets.Select(o => o.TopicPartition)
                             .ToImmutableHashSet();
                         CheckOverlappingRequests("AssignWithOffset", Sender, topicPartitions);
-
-                        var previousAssigned = _consumer.Assignment
-                            .Select(c => new TopicPartitionOffset(c, Offset.Stored));
-                        _consumer.Assign(assignWithOffset.TopicPartitionOffsets.Union(previousAssigned));
+                        
+                        _consumer.IncrementalAssign(assignWithOffset.TopicPartitionOffsets);
```

Comment:
`IncrementalAssign` is additive, instead of a replacement, which is what `Assign` does - therefore we don't need to try to re-calculate the `Offset` position of the previously assigned partitions. This helpfully eliminates a potential source of bugs for us.